### ISSUE TITLE
No way to disable max-line-length rule

### ISSUE THREAD

**Issue by jzempel** [stylelint/stylelint#581]
Once it has been set, I'm unable to disable `max-line-length`. I've tried the following:

```css
/* stylelint-disable max-line-length */
.class { foo: with-a-value-that-exceeds-max-line-length; }
/* stylelint-enable */
```

also

```css
/* stylelint-disable */
.class { foo: with-a-value-that-exceeds-max-line-length; }
/* stylelint-enable */
```

and, finally

```json
{
  "extends": "stylelint-config-custom",
  "rules": {
    "max-line-length": null
  }
}
```

where `stylelint-config-custom` defines a `"max-line-length": 80` rule. I'm using stylelint version 3.1.2.

**Comment by davidtheclark** [comment]
Are you able to disable other rules just fine, and this one is the exception?

**Comment by jzempel** [comment]
That's correct. For example, I'm using:

```css
/* stylelint-disable declaration-no-important */
.bg-blue { background-color: blue !important; }
/* stylelint-enable*/
```

and it works as expected.

**Comment by davidtheclark** [comment]
Thanks for the report, @jzempel. The problem was rooted in the fact that max-line-length attaches its warning to the root node instead of to a specific one. Because of that, I'll bet the same problem also applies to the following rules (and only them):

- max-empty-lines
- no-eol-whitespace
- no-missing-eof-newline
- string-quotes

Pushing a fix soon.

**Comment by davidtheclark** [comment]
Released in 3.1.3

**Comment by jzempel** [comment]
Thanks for the quick fix!
